src/components/Form/docs/Input.stories.mdx

```diff
@@ -1,80 +1,171 @@
+
+Don't use extra punctuation at the end of labels such as colons, periods or hyphens.
+
+If your label needs to be a question, add a question mark at the end. Avoid mixing question and statement label types within the same form where possible.
+
+Do not include links within labels or add extra visual styles/treatments.
+
+Only use an acronym in a label if it has been described separately, before the form. Avoid the introduction and description of new acronyms within a form label where possible. If an acronym has not been previously described, us its full form in the input label and describe it later if relevant.
+
+## Handling errors
 
-## Interactions
+<Canvas>
+	<Story name="Error state">
+		<Form.Text required label="Dataset name" description="Enter a dataset name" hasError value="" />
+	</Story>
+</Canvas>
+
+Error messages should follow [error message guidance](). Be concise, consistent and specific.
+
+#### If the input is empty
+
+"Enter a dataset name"
+
+#### If the input is too long
 
-N/C
+"Dataset name must be 50 characters or fewer"
 
-## Usage
+#### If the input includes characters that are not allowed
+
+"Dataset name must only include letter, number, hyphens and spaces"
+
+## Try this component
+
+<Canvas>
+	<Story
+		name="Playground"
+		args={{
+			required: false,
+			disabled: false,
+			readOnly: false,
+			hasError: false,
+			hasWarning: false,
+			label: 'Dataset name',
+			description: '',
+			value: 'Customer Data',
+		}}
+	>
+		{({
+			description,
+			disabled,
+			hasError,
+			hasWarning,
+			label,
+			readOnly,
+			required,
+			value,
+			...rest
+		}) => (
+			<Form.Text
+				disabled={disabled}
+				readOnly={readOnly}
+				hasError={hasError}
+				hasWarning={hasWarning}
+				required={required}
+				label={label}
+				value={value}
+				description={description}
+			/>
+		)}
+	</Story>
+</Canvas>
 
-N/C
+<ArgsTable story="Playground" />
+
+## Zoning
+
+<FigmaImage
+	src="https://www.figma.com/file/qc8oUSyVJM67gSsbTShftt/%F0%9F%93%90-Zoning?node-id=245%3A219"
+	alt="zoning image for inputs"
+/>
 
 ## Accessibility
 
-Press <kbd>TABS</kbd> to focus on an input
+### Features
+
+Input descriptions include the `aria-live` property and will request that screen readers announce changes to their content. Changes are given the `polite` status and so will only be announced when the screen reader next becomes idle.
+
+### Considerations
+
+Large groups of text inputs can become visually overwhelming, especially when they share a consistent length. While large groups of inputs should generally be avoided, it is advised to use content hierarchy and relevant input lengths to overcome this issue.
+
+## Research on this component
+
+As an input for non-specific information, you should be prepared for your users to enter a diverse range of content in this type of input. It can help to provide a description or example of common usage but you should not expect your user to follow this content.
```

Comment:
I like the idea of pushing for more research references even if I don't see the added value for an `<input type="text">`